### DISCUSSION TITLE
How to manually start an Incoming channel?

### DISCUSSION THREAD

**Discussion by yazinnnn** [smallrye/smallrye-reactive-messaging#2107] [Q&A]
On application startup, i should do some initialization work, then manually start the incoming channel

in spring, i can do it like this
```kotlin
fun init(){
  doSomething()
  registry.getListenerContainer("listener").start()
}

@Listener(id="listener", autoStartup="false")
fun listener(message:String){}
```

how can I do it with quarkus-smallrye-reactive-messaging-rabbitmq ?

**Comment by cescoffier** [comment]
If you use Quarkus, you can use startup event with a priority higher than _framework_.
